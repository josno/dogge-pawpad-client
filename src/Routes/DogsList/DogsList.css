.dog-img-container {
	width: 100%;
	height: 200px;
	box-shadow: 10px 10px #5d576b;
	border: 2px solid black;
}

.dog-img-container:active {
	box-shadow: none;
}

.dog-item {
	margin: 20px;
}

.add-a-dog-button {
	font-size: 1em;
	border: none;
}

.add-a-dog-button:hover {
	cursor: pointer;
}

.dog-list-container {
	padding-top: 40px;
}

.dogs-list {
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
}

.list-container {
	padding-top: 30px;
}

.dog-list-container {
	display: flex;
	flex-wrap: wrap;
	justify-content: space-evenly;
	height: 80vh;
}

.dogs-list-name {
	color: #fcfcfc;
	background-color: #f7567c;
	font-size: 24px;
	font-weight: bolder;
	height: 40px;
	line-height: 40px;
	position: relative;
	top: -71px;
}
