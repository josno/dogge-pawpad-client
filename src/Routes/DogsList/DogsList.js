import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import DogListItem from '../../Components/DogListItem/DogListItem';
import DogsApiService from '../../services/api-service';
import './DogsList.css';

class DogList extends Component {
	constructor(props) {
		super(props);
		this.state = {
			error: null,
			dogs: []
		};
	}

	componentDidMount() {
		DogsApiService.getDogs().then(responsejson => {
			if (responsejson.length === 0) {
				this.setState({
					error: 'Something went wrong, try again'
				});
			}

			this.setState({
				dogs: [...responsejson]
			});
		});
	}

	render() {
		const { dogs } = this.state;

		return (
			<main className="DogList">
				<section>
					<div className="list-container">
						<h1 className="dogs-list-title"> Current Dogs </h1>
					</div>
					<div className="dogs-list">
						{dogs.map(d => (
							<DogListItem
								name={d.dog_name}
								id={d.id}
								key={d.id}
								img={d.profile_img}
							/>
						))}

						<li className="dog-item">
							<Link className="dog-img-link" to={'/add-new-dog'}>
								<div className="dog-img-container add-dog">
									<button className="add-a-dog-button">
										<h2>
											Add A <br /> New Dog
										</h2>
									</button>
								</div>
							</Link>
						</li>
					</div>
				</section>
			</main>
		);
	}
}

export default DogList;
