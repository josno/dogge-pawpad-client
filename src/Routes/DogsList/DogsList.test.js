import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import DogsList from './DogsList';

describe('<DogsList />', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<DogsList />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<DogsList />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
