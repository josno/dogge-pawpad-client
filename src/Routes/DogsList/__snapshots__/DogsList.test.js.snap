// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogsList /> renders as expected 1`] = `
<main
  className="DogList"
>
  <section>
    <div
      className="list-container"
    >
      <h1
        className="dogs-list-title"
      >
         Current Dogs 
      </h1>
    </div>
    <div
      className="dogs-list"
    >
      <li
        className="dog-item"
      >
        <a
          className="dog-img-link"
          href="/add-new-dog"
          onClick={[Function]}
        >
          <div
            className="dog-img-container add-dog"
          >
            <button
              className="add-a-dog-button"
            >
              <h2>
                Add A 
                <br />
                 New Dog
              </h2>
            </button>
          </div>
        </a>
      </li>
    </div>
  </section>
</main>
`;
